### PR TITLE
Use PathType for file paths

### PR DESCRIPTION
**Pull request by blowekamp**
Introduce PathType where std::string is used to enable easy change of the type to std::filesystem::path.

### CODE REVIEW

File: Code/Common/include/sitkMacro.h [hunk 1]
```diff
@@ -64,6 +64,13 @@
 
 namespace itk::simple {
 
+
+#if defined(SITK_USE_STD_FILESYSTEM_PATH)
+using PathType = std::filesystem::path;
+#else
+using PathType = std::string;
+#endif
```
[the remark this answers is not in the document]
Comment (blowekamp) [reply Code/Common/include/sitkMacro.h]: Created separate header file

**Comment by blowekamp** [comment]
@N-Dekker  Do you have any thoughts on std::filesystem::path? This PR is an intermediate step to transition for SimpleITK. It should be useful to help detect ( maybe one day support ) non-ASCII paths.

**Comment by N-Dekker** [comment]
> Do you have any thoughts on `std::filesystem::path`? This PR is an intermediate step to transition for SimpleITK. It should be useful to help detect ( maybe one day support ) non-ASCII paths.

I think `std::filesystem` is cool. Unfortunately we cannot yet use it in ITK, unless we drop GCC 7 support, as we discussed last year at https://discourse.itk.org/t/shall-we-start-using-c-17-at-the-master-branch/5828

However, I don't think that `std::filesystem` specifically addressed non-ASCII paths. Non-ASCII paths are _only_ a Windows problem, right? I mean, non-ASCII paths do work fine on Linux and MacOS, right?

On Windows, there is a Region option for users to switch on "Use Unicode UTF-8 for worldwide language support". I _think_ that should enable Unicode characters in file paths, represented by either `std::string` or `std::filesystem::path`. 

My 2 cent!

**Comment by blowekamp** [comment]
> However, I don't think that `std::filesystem` specifically addressed non-ASCII paths. Non-ASCII paths are _only_ a Windows problem, right? I mean, non-ASCII paths do work fine on Linux and MacOS, right?

From my reading, it seams std::filesystem is represented as wchar on Windows, and just 8-bit char on other systems. Also there being OS level methods to open files with wchar strings. So I was seeing it as an operating system dependent way to define paths.

Additionally SWIG added some nice wrapping of std::filesystem::path to pathlib.Path types for Python. 

> Unfortunately we cannot yet use it in ITK, unless we drop GCC 7 support, as we discussed last year at https://discourse.itk.org/t/shall-we-start-using-c-17-at-the-master-branch/5828

The big hold out there was Ubuntu 18.04, which is now EOL'd. Not sure what the schedule is for ITK releases any more.

**Comment by N-Dekker** [comment]
>> Unfortunately we cannot yet use it in ITK, unless we drop GCC 7 support, as we discussed last year at https://discourse.itk.org/t/shall-we-start-using-c-17-at-the-master-branch/5828
> 
> The big hold out there was Ubuntu 18.04, which is now EOL'd. Not sure what the schedule is for ITK releases any more.

Good point! I just asked if the time is right for ITK to move forward to GCC >= 9.0, at https://discourse.itk.org/t/shall-we-start-using-c-17-at-the-master-branch/5828/18?u=niels_dekker

**Review by zivy — APPROVED** [review]

**Pull request merged**